consensus/src/persistent_liveness_storage.rs

```diff
@@ -150,23 +158,71 @@ impl LedgerRecoveryData {
         } else {
             let root_ordered_cert = quorum_certs
                 .iter()
-                .find(|qc| qc.commit_info().id() == root_block.id())
-                .ok_or_else(|| format_err!("No LI found for root: {}", root_id))?
+                .find(|qc| qc.commit_info().id() == commit_block.id())
+                .ok_or_else(|| format_err!("No LI found for root: {}", latest_commit_id))?
                 .clone()
                 .into_wrapped_ledger_info();
             let root_commit_cert = root_ordered_cert
                 .create_merged_with_executed_state(latest_ledger_info_sig)
                 .expect("Inconsistent commit proof and evaluation decision, cannot commit block");
             (root_ordered_cert, root_commit_cert)
         };
-        info!("Consensus root block is {}", root_block);
-
-        Ok(RootInfo(
-            Box::new(root_block),
-            root_quorum_cert,
-            root_ordered_cert,
-            root_commit_cert,
-        ))
+
+        let window_start_round = blocks[latest_commit_idx]
```

Comment:
This round may not necessarily exist right? Probably timed out. Also, some blocks between can be missing too. So, this may not be the actual start round? 

Do we calculate the window based on consecutive round numbers? or do we calculate based on successful blocks?

Reply:
I think window size refers to round number strictly, not number of blocks, not consecutive rounds.